
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Predict &#8212; CELLxGENE Deep Dive Course</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=796348d33e8b1d947c94" rel="stylesheet">
<link href="../_static/styles/bootstrap.css?digest=796348d33e8b1d947c94" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=796348d33e8b1d947c94" rel="stylesheet">

  
  <link href="../_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=796348d33e8b1d947c94" rel="stylesheet">
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2">
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2">
<link rel="preload" as="font" type="font/woff2" crossorigin href="../_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=4ec06e9971c5264fbd345897d5258098f11cc577" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.4045f2051d55cab465a707391d5b2007.min.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/bootstrap.js?digest=796348d33e8b1d947c94">
<link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=796348d33e8b1d947c94">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=8bf782fb4ee92b3d3646425e50f299c4e1fd152d"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'annotate/predict';</script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="CZ CELLxGENE Discover Census" href="../census/census.html" />
    <link rel="prev" title="Annotate Data" href="annotate.html" />
    <script async defer data-domain="czi-napari-segmentation-workshop" src="https://plausible.io/js/plausible.js"></script>
    
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="docsearch:language" content="None">

  </head>
  
  
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="180" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>

  
  <input type="checkbox" class="sidebar-toggle" name="__primary" id="__primary">
  <label class="overlay overlay-primary" for="__primary"></label>

  
  <input type="checkbox" class="sidebar-toggle" name="__secondary" id="__secondary">
  <label class="overlay overlay-secondary" for="__secondary"></label>

  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
      
<form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false">
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
    </div>
  </div>

  
  <nav class="bd-header navbar navbar-expand-lg bd-navbar" id="navbar-main"><div class="bd-header__inner bd-page-width">
  <label class="sidebar-toggle primary-toggle" for="__primary">
      <span class="fa-solid fa-bars"></span>
  </label>
  <div id="navbar-start">
    
    
  


<a class="navbar-brand logo" href="../intro.html">

  
  
  
  
  
  
  

  
    <img src="../_static/cellxgeneLogoFull.webp" class="logo__image only-light" alt="Logo image">
    <img src="../_static/cellxgeneLogoFull.webp" class="logo__image only-dark" alt="Logo image">
  
  
</a>
    
  </div>

  
  <div class="col-lg-9 navbar-header-items">
    <div id="navbar-center" class="mr-auto">
      
      <div class="navbar-center-item">
        <nav class="navbar-nav">
    <p class="sidebar-header-items__title" role="heading" aria-level="1" aria-label="Site Navigation">
        Site Navigation
    </p>
    <ul id="navbar-main-elements" class="navbar-nav">
        
                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../preface/about.html">
                        About this Project
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../preface/whomadethis.html">
                        About Me
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../preface/feedback.html">
                        Frequently Asked Questions
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../dataportal/01_discoverportal.html">
                        Discover single cell and spatial data
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../geneExpression/geneExpression.html">
                        Discover Patterns In Gene Expression
                      </a>
                    </li>
                
            <div class="nav-item dropdown">
                <button class="btn dropdown-toggle nav-item" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    More
                </button>
                <div class="dropdown-menu">
                    
                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../explorer/explorer.html">
                        Explore single cell datasets
                      </a>
                    </li>
                

                    <li class="nav-item current active">
                      <a class="nav-link nav-internal" href="overview.html">
                        Annotate and Analyze Your Data
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../census/census.html">
                        CZ CELLxGENE Discover Census
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../closing.html">
                        Closing
                      </a>
                    </li>
                
                </div>
            </div>
            
    </ul>
</nav>
      </div>
      
    </div>

    <div id="navbar-end">
      
        <div class="navbar-end-item navbar-persistent--container">
          
<button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-toggle="tooltip">
  <i class="fa-solid fa-magnifying-glass"></i>
</button>
        </div>
      
      
      <div class="navbar-end-item">
        <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
</button>
      </div>
      
      <div class="navbar-end-item">
        <ul id="navbar-icon-links" class="navbar-nav" aria-label="Icon Links">
      </ul>
      </div>
      
    </div>
  </div>


  
  
    <div class="navbar-persistent--mobile">
<button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-toggle="tooltip">
  <i class="fa-solid fa-magnifying-glass"></i>
</button>
    </div>
  

  
  <label class="sidebar-toggle secondary-toggle" for="__secondary">
      <span class="fa-solid fa-outdent"></span>
  </label>
  

</div>
  </nav>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        
  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
      <div class="sidebar-header-items__center">
      
      <div class="navbar-center-item">
        <nav class="navbar-nav">
    <p class="sidebar-header-items__title" role="heading" aria-level="1" aria-label="Site Navigation">
        Site Navigation
    </p>
    <ul id="navbar-main-elements" class="navbar-nav">
        
                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../preface/about.html">
                        About this Project
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../preface/whomadethis.html">
                        About Me
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../preface/feedback.html">
                        Frequently Asked Questions
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../dataportal/01_discoverportal.html">
                        Discover single cell and spatial data
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../geneExpression/geneExpression.html">
                        Discover Patterns In Gene Expression
                      </a>
                    </li>
                
            <div class="nav-item dropdown">
                <button class="btn dropdown-toggle nav-item" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    More
                </button>
                <div class="dropdown-menu">
                    
                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../explorer/explorer.html">
                        Explore single cell datasets
                      </a>
                    </li>
                

                    <li class="nav-item current active">
                      <a class="nav-link nav-internal" href="overview.html">
                        Annotate and Analyze Your Data
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../census/census.html">
                        CZ CELLxGENE Discover Census
                      </a>
                    </li>
                

                    <li class="nav-item">
                      <a class="nav-link nav-internal" href="../closing.html">
                        Closing
                      </a>
                    </li>
                
                </div>
            </div>
            
    </ul>
</nav>
      </div>
      
      </div>
    

    
    
    <div class="sidebar-header-items__end">
      
      <div class="navbar-end-item">
        <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
</button>
      </div>
      
      <div class="navbar-end-item">
        <ul id="navbar-icon-links" class="navbar-nav" aria-label="Icon Links">
      </ul>
      </div>
      
    </div>
    
  </div>

  
  <div class="sidebar-start-items sidebar-primary__section">
    <div class="sidebar-start-items__item">
  


<a class="navbar-brand logo" href="../intro.html">

  
  
  
  
  
  
  

  
    <img src="../_static/cellxgeneLogoFull.webp" class="logo__image only-light" alt="Logo image">
    <img src="../_static/cellxgeneLogoFull.webp" class="logo__image only-dark" alt="Logo image">
  
  
</a>
    </div>
    <div class="sidebar-start-items__item">
<form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false">
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
    </div>
    <div class="sidebar-start-items__item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    CELLxGENE Suite Deep Dive
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Preface</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../preface/about.html">About this Project</a></li>
<li class="toctree-l1"><a class="reference internal" href="../preface/whomadethis.html">About Me</a></li>


<li class="toctree-l1"><a class="reference internal" href="../preface/feedback.html">Frequently Asked Questions</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">CZ CELLxGENE Discover</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="../dataportal/01_discoverportal.html">Discover single cell and spatial data</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../dataportal/portal.html">Discover UI Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../dataportal/submission.html">Summary of CZ CELLxGENE Discover Submission Requirements</a></li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../geneExpression/geneExpression.html">Discover Patterns In Gene Expression</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../geneExpression/interpretdotplot.html">Query and Visualize Gene Expression Across Tissues</a></li>
<li class="toctree-l2"><a class="reference internal" href="../geneExpression/cellgeneordering.html">Cell Type and Gene ordering</a></li>


<li class="toctree-l2"><a class="reference internal" href="../geneExpression/dataprocessing.html">Gene Expression Data Processing</a></li>

<li class="toctree-l2"><a class="reference internal" href="../geneExpression/tissues.html">Available Tissues in Gene Expression</a></li>
<li class="toctree-l2"><a class="reference internal" href="../geneExpression/findmarkergenes.html">Find Marker Genes</a></li>
<li class="toctree-l2 has-children"><a class="reference internal" href="../cellguide/cellguide.html">Explore relationships between cell types</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="../cellguide/ontologytree.html">Understanding how cell types are related using the CL ontology</a></li>
<li class="toctree-l3"><a class="reference internal" href="../cellguide/celltype.html">Cell type descriptions, relationships, markers and data</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../explorer/explorer.html">Explore single cell datasets</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2"><a class="reference internal" href="../explorer/UI-Overview.html">Explorer UI Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="../explorer/metadata.html">Visualize gene expression, categorical metadata, and quality control metrics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../explorer/subset.html">Filter and subset cells based on stackable metadata filters</a></li>
<li class="toctree-l2"><a class="reference internal" href="../explorer/diffexp.html">Differential expression</a></li>
</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">CZ CELLxGENE Annotate</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active has-children"><a class="reference internal" href="overview.html">Annotate and Analyze Your Data</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-5"><i class="fa-solid fa-chevron-down"></i></label><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="start.html">Prepare Data for CELLxGENE Annotate</a></li>

<li class="toctree-l2"><a class="reference internal" href="annotate.html">Annotate Data</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Predict</a></li>

</ul>
</li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Discover Census and API</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../census/census.html">CZ CELLxGENE Discover Census</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">Closing</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../closing.html">Closing</a></li>
</ul>

    </div>
</nav>
    </div>
  </div>
  

  
  <div class="sidebar-end-items sidebar-primary__section">
    <div class="sidebar-end-items__item">
    </div>
  </div>

  
  <div id="rtd-footer-container"></div>

      </div>
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

        <div class="bd-content">
          <div class="bd-article-container">
            
            <div class="bd-header-article">
                



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        <label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" data-toggle="tooltip" data-placement="right" title="Toggle primary sidebar">
            <span class="fa-solid fa-bars"></span>
        </label>
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="btn btn-sm"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<div class="dropdown dropdown-repository-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      <li><a href="https://github.com/MaximilianLombardo/cxg-deep-dive-workshop" target="_blank"
   class="btn btn-sm dropdown-item"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">repository</span>
</a>
</a>
      
      <li><a href="https://github.com/MaximilianLombardo/cxg-deep-dive-workshop/issues/new?title=Issue%20on%20page%20%2Fannotate/predict.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm dropdown-item"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">open issue</span>
</a>
</a>
      
  </ul>
</div>



<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      <li><a href="../_sources/annotate/predict.md" target="_blank"
   class="btn btn-sm dropdown-item"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</a>
      
      <li>
<button onclick="printPdf(this)"
  class="btn btn-sm dropdown-item"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</a>
      
  </ul>
</div>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary" data-toggle="tooltip" data-placement="left" title="Toggle secondary sidebar">
            <span class="fa-solid fa-list"></span>
        </label>
    </div>
</div>
            </div>
            
            

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Predict</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Predict
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#automated-annotation-of-single-cell-data-experimental">
   Automated Annotation of Single Cell Data (EXPERIMENTAL)
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#installation-of-cellxgene-annotate">
     Installation of CELLxGENE[annotate]
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#option-1-conda-setup">
       Option 1: conda setup
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#option-2-venv-setup">
       Option 2: venv setup
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cellxgene-annotation-workflow">
     CELLxGENE Annotation Workflow
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#quickstart">
       Quickstart
      </a>
      <ul class="nav section-nav flex-column">
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#conda">
         conda
        </a>
       </li>
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#venv">
         venv
        </a>
       </li>
      </ul>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-0-annotation-workflow-inputs">
       Step 0: Annotation workflow inputs
      </a>
      <ul class="nav section-nav flex-column">
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#query-data">
         Query data
        </a>
       </li>
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#prediction-model">
         Prediction model
        </a>
       </li>
      </ul>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-1-running-cellxgene-annotate">
       Step 1: Running
       <code class="docutils literal notranslate">
        <span class="pre">
         cellxgene
        </span>
        <span class="pre">
         annotate
        </span>
       </code>
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-2-explore-and-refine">
       Step 2: Explore and Refine
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#useful-links">
       Useful Links
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>

            <article class="bd-article" role="main">
              
  <section class="tex2jax_ignore mathjax_ignore" id="predict">
<h1>Predict<a class="headerlink" href="#predict" title="Permalink to this headline">#</a></h1>
</section>
<section class="tex2jax_ignore mathjax_ignore" id="automated-annotation-of-single-cell-data-experimental">
<h1>Automated Annotation of Single Cell Data (EXPERIMENTAL)<a class="headerlink" href="#automated-annotation-of-single-cell-data-experimental" title="Permalink to this headline">#</a></h1>
<p>CZ CELLxGENE Annotate enables automatic cell type annotation of newly generated single cell datasets using pre-trained models. Annotation augments your dataset with the following new information:</p>
<ul class="simple">
<li><p>transferred cell type labels (a.k.a. the most probable cell type)</p></li>
<li><p>uncertainty scores (measure of the confidence of the predicted label - takes on a value between 0 and 1 [0 = confident, 1 = uncertain])</p></li>
<li><p>a projection of the query dataset into the reference embedding, and a newly computed UMAP based on that projection</p></li>
</ul>
<p>After you complete the automatic annotation workflow, you can explore and refine your predictions using <a class="reference external" href="https://cellxgene.cziscience.com/docs/04__Analyze%20Public%20Data/4_1__Hosted%20Tutorials">standard features in CELLxGENE</a> such as:</p>
<ul class="simple">
<li><p>visualizing clusters by predicted cell type</p></li>
<li><p>plotting marker genes</p></li>
<li><p>performing differential expression</p></li>
<li><p>refining cell type labels by creating custom annotations</p></li>
</ul>
<p><strong>Disclaimer:</strong> The Annotation feature in CELLxGENE is new and experimental. We intend for the results generated by the workflow to be a starting point for annotating your single cell dataset. If you run into issues completing this workflow, we would love to hear about your experience. Feel free to reach out to <span class="xref myst">cellxgene&#64;chanzuckerberg.com</span> to get help running the workflow or to forward constructive feedback. In the meantime, in order to obtain the best results, we make the following recommendations:</p>
<ul class="simple">
<li><p>use this predictive annotation workflow with data generated by 10X platforms</p></li>
<li><p>use the tissue model which is most closely related to the tissue profiled in your single cell experiment (currently available models are: lung, pan-tissue immune)</p></li>
</ul>
<section id="installation-of-cellxgene-annotate">
<h2>Installation of CELLxGENE[annotate]<a class="headerlink" href="#installation-of-cellxgene-annotate" title="Permalink to this headline">#</a></h2>
<p>You can install the latest experimental release of CELLxGENE Annotate to gain access to the automatic annotation subcommand. It is highly recommended to perform installation in a clean Python environment using <a class="reference external" href="https://docs.conda.io/en/latest/">Conda</a> or <a class="reference external" href="https://docs.python.org/3/library/venv.html">venv</a> with Python version 3.9+. Find examples of virtual environment setup using venv and conda below.</p>
<p>CELLxGENE[annotate] is currently only tested on OS X and Ubuntu operating systems.</p>
<section id="option-1-conda-setup">
<h3>Option 1: conda setup<a class="headerlink" href="#option-1-conda-setup" title="Permalink to this headline">#</a></h3>
<p>To install <code class="docutils literal notranslate"><span class="pre">conda</span></code>, download and install from one of the distributions:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.anaconda.com/products/distribution">Anaconda</a></p></li>
<li><p><a class="reference external" href="https://docs.conda.io/en/latest/miniconda.html">Miniconda</a></p></li>
<li><p><a class="reference external" href="https://github.com/conda-forge/miniforge">Miniforge</a></p></li>
</ul>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">create</span> <span class="o">-</span><span class="n">n</span> <span class="n">cellxgene</span> <span class="s1">&#39;python=3.9&#39;</span>

<span class="n">conda</span> <span class="n">activate</span> <span class="n">cellxgene</span>

<span class="n">pip</span> <span class="n">install</span> <span class="s1">&#39;cellxgene[annotate]&#39;</span>
</pre></div>
</div>
<p>To test the success of your installation, run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cellxgene</span> <span class="n">annotate</span> <span class="o">--</span><span class="n">help</span>

</pre></div>
</div>
</section>
<section id="option-2-venv-setup">
<h3>Option 2: venv setup<a class="headerlink" href="#option-2-venv-setup" title="Permalink to this headline">#</a></h3>
<p>For an overview of <code class="docutils literal notranslate"><span class="pre">venv</span></code>, refer to the link in the above section.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="o">-</span><span class="n">m</span> <span class="n">venv</span> <span class="n">cxg_annotate</span>  <span class="c1"># create a virtual environment in a new the current directory</span>

<span class="n">source</span> <span class="n">cxg_annotate</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">activate</span>

<span class="n">pip</span> <span class="n">install</span> <span class="s1">&#39;cellxgene[annotate]&#39;</span>
</pre></div>
</div>
<p>If <code class="docutils literal notranslate"><span class="pre">conda</span></code> is not installed on your system, install these additional packages:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">scvi</span><span class="o">-</span><span class="n">tools</span><span class="o">==</span><span class="mf">0.16.2</span> <span class="s1">&#39;scanpy[leiden]==1.9.3&#39;</span> <span class="n">xgboost</span><span class="o">==</span><span class="mf">1.6.1</span> <span class="n">torch</span><span class="o">==</span><span class="mf">1.11.0</span>
</pre></div>
</div>
<p>If this fails, we recommend you use a Conda environment (see the <span class="xref myst">conda setup</span> section, above)</p>
<p>To test the success of your installation, run:</p>
<p><code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span> <span class="pre">--help</span></code></p>
</section>
</section>
<section id="cellxgene-annotation-workflow">
<h2>CELLxGENE Annotation Workflow<a class="headerlink" href="#cellxgene-annotation-workflow" title="Permalink to this headline">#</a></h2>
<p>See our quickstart guide to quickly see if you can run through the pipeline or check out the more detailed sections below for more detailed descriptions of the workflow. Both assume that you have properly set up a clean virtual environment to work from.</p>
<section id="quickstart">
<h3>Quickstart<a class="headerlink" href="#quickstart" title="Permalink to this headline">#</a></h3>
<p>In this quickstart, we will use our lung model (trained on the <a class="reference external" href="https://cellxgene.cziscience.com/collections/6f6d381a-7701-4781-935c-db10d30de293">Integrated Human Lung Cell atlas</a>), to make annotations on a subset of another human lung dataset, <a class="reference external" href="https://cellxgene.cziscience.com/collections/625f6bf4-2f33-4942-962e-35243d284837">LungMap</a>. You can download the entire LungMap dataset <a class="reference external" href="https://cellxgene.cziscience.com/collections/625f6bf4-2f33-4942-962e-35243d284837">here</a>. Depending on your environment manager, execute one of the following quickstart commands:</p>
<section id="conda">
<h4>conda<a class="headerlink" href="#conda" title="Permalink to this headline">#</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># create a directory to download data, run annotation pipeline and store annotated object</span>
<span class="n">mkdir</span> <span class="n">annotations</span>

<span class="c1"># navigate to newly created directory</span>
<span class="n">cd</span> <span class="n">annotations</span>

<span class="c1"># retrieve sample dataset (alternatively, you can use curl to retrieve the file)</span>
<span class="c1"># ex: curl https://cellxgene-annotation-public.s3.us-west-2.amazonaws.com/cell_type/tutorial/minilung.h5ad &gt; minilung.h5ad</span>
<span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">cellxgene</span><span class="o">-</span><span class="n">annotation</span><span class="o">-</span><span class="n">public</span><span class="o">.</span><span class="n">s3</span><span class="o">.</span><span class="n">us</span><span class="o">-</span><span class="n">west</span><span class="o">-</span><span class="mf">2.</span><span class="n">amazonaws</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">cell_type</span><span class="o">/</span><span class="n">tutorial</span><span class="o">/</span><span class="n">minilung</span><span class="o">.</span><span class="n">h5ad</span>


<span class="c1"># generate annotations</span>
<span class="n">cellxgene</span> <span class="n">annotate</span> <span class="n">minilung</span><span class="o">.</span><span class="n">h5ad</span>  <span class="o">-</span><span class="n">m</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">cellxgene</span><span class="o">-</span><span class="n">annotation</span><span class="o">-</span><span class="n">public</span><span class="o">.</span><span class="n">s3</span><span class="o">.</span><span class="n">us</span><span class="o">-</span><span class="n">west</span><span class="o">-</span><span class="mf">2.</span><span class="n">amazonaws</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">cell_type</span><span class="o">/</span><span class="n">models</span><span class="o">/</span><span class="n">hlca_20220920223732</span><span class="o">.</span><span class="n">zip</span> <span class="o">-</span><span class="n">o</span> <span class="n">minilungAnnotated</span><span class="o">.</span><span class="n">h5ad</span> <span class="o">--</span><span class="n">mlflow</span><span class="o">-</span><span class="n">env</span><span class="o">-</span><span class="n">manager</span> <span class="n">conda</span> <span class="o">--</span><span class="n">no</span><span class="o">-</span><span class="n">use</span><span class="o">-</span><span class="n">gpu</span> <span class="o">--</span><span class="n">gene</span><span class="o">-</span><span class="n">column</span><span class="o">-</span><span class="n">name</span> <span class="n">feature_name</span>

<span class="c1"># launch cellxgene explorer</span>
<span class="n">cellxgene</span> <span class="n">launch</span> <span class="n">minilungAnnotated</span><span class="o">.</span><span class="n">h5ad</span>
</pre></div>
</div>
</section>
<section id="venv">
<h4>venv<a class="headerlink" href="#venv" title="Permalink to this headline">#</a></h4>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># create a directory to download data, run annotation pipeline and store annotated object</span>
<span class="n">mkdir</span> <span class="n">annotations</span>

<span class="c1"># navigate to newly created directory</span>
<span class="n">cd</span> <span class="n">annotations</span>

<span class="c1"># retrieve sample dataset (alternatively, you can use curl to retrieve the file)</span>
<span class="c1"># ex: curl https://cellxgene-annotation-public.s3.us-west-2.amazonaws.com/cell_type/tutorial/minilung.h5ad &gt; minilung.h5ad</span>
<span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">cellxgene</span><span class="o">-</span><span class="n">annotation</span><span class="o">-</span><span class="n">public</span><span class="o">.</span><span class="n">s3</span><span class="o">.</span><span class="n">us</span><span class="o">-</span><span class="n">west</span><span class="o">-</span><span class="mf">2.</span><span class="n">amazonaws</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">cell_type</span><span class="o">/</span><span class="n">tutorial</span><span class="o">/</span><span class="n">minilung</span><span class="o">.</span><span class="n">h5ad</span>


<span class="c1"># generate annotations</span>
<span class="n">cellxgene</span> <span class="n">annotate</span> <span class="n">minilung</span><span class="o">.</span><span class="n">h5ad</span> <span class="o">-</span><span class="n">m</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">cellxgene</span><span class="o">-</span><span class="n">annotation</span><span class="o">-</span><span class="n">public</span><span class="o">.</span><span class="n">s3</span><span class="o">.</span><span class="n">us</span><span class="o">-</span><span class="n">west</span><span class="o">-</span><span class="mf">2.</span><span class="n">amazonaws</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">cell_type</span><span class="o">/</span><span class="n">models</span><span class="o">/</span><span class="n">hlca_20220920223732</span><span class="o">.</span><span class="n">zip</span> <span class="o">-</span><span class="n">o</span> <span class="n">minilungAnnotated</span><span class="o">.</span><span class="n">h5ad</span> <span class="o">--</span><span class="n">mlflow</span><span class="o">-</span><span class="n">env</span><span class="o">-</span><span class="n">manager</span> <span class="n">local</span> <span class="o">--</span><span class="n">no</span><span class="o">-</span><span class="n">use</span><span class="o">-</span><span class="n">gpu</span> <span class="o">--</span><span class="n">gene</span><span class="o">-</span><span class="n">column</span><span class="o">-</span><span class="n">name</span> <span class="n">feature_name</span>

<span class="c1"># launch cellxgene explorer</span>
<span class="n">cellxgene</span> <span class="n">launch</span> <span class="n">minilungAnnotated</span><span class="o">.</span><span class="n">h5ad</span>
</pre></div>
</div>
<p>By default, you can find annotations in the <code class="docutils literal notranslate"><span class="pre">obs</span></code> dataframe underneath the names: <code class="docutils literal notranslate"><span class="pre">cxg_cell_type_predicted</span></code> for cell type labels and <code class="docutils literal notranslate"><span class="pre">cxg_cell_type_predicted_uncertainty</span></code> for corresponding uncertainty scores.</p>
<p>You can find an already annotated version of the LungMap subset <a class="reference external" href="https://cellxgene-annotation-public.s3.us-west-2.amazonaws.com/cell_type/tutorial/minilung_cxg_annotated.h5ad">here</a>.</p>
</section>
</section>
<section id="step-0-annotation-workflow-inputs">
<h3>Step 0: Annotation workflow inputs<a class="headerlink" href="#step-0-annotation-workflow-inputs" title="Permalink to this headline">#</a></h3>
<p>CELLxGENE’s automated annotation has two essential inputs:</p>
<ul class="simple">
<li><p>a query dataset to be annotated (structured as an <code class="docutils literal notranslate"><span class="pre">anndata</span></code> object (read more about the <code class="docutils literal notranslate"><span class="pre">anndata</span></code> object <a class="reference external" href="https://anndata.readthedocs.io/en/latest/">here</a>))</p></li>
<li><p>a model that will be used to generate the annotations</p></li>
</ul>
<section id="query-data">
<h4>Query data<a class="headerlink" href="#query-data" title="Permalink to this headline">#</a></h4>
<p>You should create an <code class="docutils literal notranslate"><span class="pre">anndata</span></code> object (written to an <code class="docutils literal notranslate"><span class="pre">h5ad</span></code> file) that is compatible with the <code class="docutils literal notranslate"><span class="pre">annotate</span></code> subcommand:</p>
<ul class="simple">
<li><p>contains raw counts in the <code class="docutils literal notranslate"><span class="pre">adata.X</span></code>, <code class="docutils literal notranslate"><span class="pre">adata.raw.X</span></code>, or in a specified layer stored in <code class="docutils literal notranslate"><span class="pre">adata.layers</span></code></p></li>
<li><p>meets the additional <a class="reference external" href="https://cellxgene.cziscience.com/docs/05__Annotate%20and%20Analyze%20Your%20Data/5_3__Preparing%20Data">requirements</a> for launching and visualizing a dataset with CELLxGENE desktop</p>
<ul>
<li><p>note that the input object does not require a visualization embedding and that one will be generated as a part of running the annotation pipeline</p></li>
<li><p>the embedding that results from running the annotation pipeline will be stored as <code class="docutils literal notranslate"><span class="pre">adata.obsm['cxg_cell_type_umap']</span></code>, however you can use the command line options <code class="docutils literal notranslate"><span class="pre">--annotation-prefix</span></code>, <code class="docutils literal notranslate"><span class="pre">--annotation-type</span></code>, and <code class="docutils literal notranslate"><span class="pre">--run-name</span></code> to manipulate the name of the resulting embedding</p></li>
</ul>
</li>
</ul>
<p><strong>Generating annotations from 10X cellranger outputs</strong></p>
<p><a class="reference external" href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/tutorial_ov">10X cellranger</a> is one of the standard pipelines for quantifying gene expression values (generating the count matrix). While the <code class="docutils literal notranslate"><span class="pre">annotate</span></code> subcommand does not currently offer the ability to read cellranger outputs directly, it is relatively easy to create a compatible <code class="docutils literal notranslate"><span class="pre">anndata</span></code> object using <code class="docutils literal notranslate"><span class="pre">scanpy</span></code>. Here are some resources to get you started:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html">ScanPy PBMC 3K clustering tutorial</a></p></li>
<li><p><a class="reference external" href="https://scanpy.readthedocs.io/en/stable/generated/scanpy.read_10x_mtx.html">scanpy.read_10x_mtx()</a></p></li>
<li><p><a class="reference external" href="https://scanpy.readthedocs.io/en/stable/generated/scanpy.read_10x_h5.html">scanpy.read_10x_h5()</a></p></li>
</ul>
<p>It is important that once you have constructed your object, that you perform initial QC filtering of low quality cells from the dataset. See this <a class="reference external" href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html">link</a> for more details (specifically, the section entitled ‘Basic Filtering’).</p>
</section>
<section id="prediction-model">
<h4>Prediction model<a class="headerlink" href="#prediction-model" title="Permalink to this headline">#</a></h4>
<p>You can download models to perform prediction for the following tissues:</p>
<p>Reference Mapping Models:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://cellxgene-annotation-public.s3.us-west-2.amazonaws.com/cell_type/models/hlca_20230314151144.zip">HLCA Lung Model</a></p></li>
<li><p><a class="reference external" href="https://cellxgene-annotation-public.s3.us-west-2.amazonaws.com/cell_type/models/immune_ref_20230314151144.zip">Pan Tissue Immune Reference</a></p></li>
</ul>
<p>Logistic Regression Models:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://cellxgene-annotation-public.s3.us-west-2.amazonaws.com/cell_type/models/hlca_lr_20230314151144.zip">HLCA Lung Model</a></p></li>
</ul>
</section>
</section>
<section id="step-1-running-cellxgene-annotate">
<h3>Step 1: Running <code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span></code><a class="headerlink" href="#step-1-running-cellxgene-annotate" title="Permalink to this headline">#</a></h3>
<p>After installing the latest version of CELLxGENE and the annotate subcommand, you can run a standard call to the command like so (see <code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span> <span class="pre">--help</span></code> for a full list of available options):</p>
<p><code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span> <span class="pre">./pathTo/data.h5ad</span> <span class="pre">--model-url</span> <span class="pre">modelURL</span> <span class="pre">--output-h5ad-file</span> <span class="pre">./pathTo/annotatedData.h5ad</span></code></p>
<p>In the call of the <code class="docutils literal notranslate"><span class="pre">annotate</span></code> subcommand above, we specify three arguments</p>
<ul class="simple">
<li><p>the local path to your query dataset (in <code class="docutils literal notranslate"><span class="pre">anndata</span></code> format)</p></li>
<li><p>the url of the tissue model which you like to use to perform reference mapping (for a list of available reference models, refer to the <span class="xref myst">section above</span>)</p></li>
<li><p>the local path where <code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span></code> will write a new <code class="docutils literal notranslate"><span class="pre">anndata</span></code> object that includes the predicted annotations</p></li>
</ul>
<p><strong>Note:</strong> the first time use of a given model will:</p>
<ul class="simple">
<li><p>trigger a one-time download of the model, stored locally for repeated use (in .model_cache by default)</p></li>
<li><p>take some extra time to setup an (internal) python env in which to run the model</p></li>
</ul>
<p>Once your call to <code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span></code> completes, you will have the following new information in your <code class="docutils literal notranslate"><span class="pre">anndata</span></code> object:</p>
<ul class="simple">
<li><p>transferred annotations/labels [<code class="docutils literal notranslate"><span class="pre">categorical</span></code>] (i.e. cell type) - these will be stored under <code class="docutils literal notranslate"><span class="pre">adata.obs['cxg_cell_type_predicted']</span></code> by default</p></li>
<li><p>confidence scores [<code class="docutils literal notranslate"><span class="pre">numeric</span></code>] - stored under <code class="docutils literal notranslate"><span class="pre">adata.obs['cxg_cell_type_predicted_uncertainty']</span></code> by default</p></li>
<li><p>a mapping into the reference embedding space - stored under <code class="docutils literal notranslate"><span class="pre">adata.obsm['cxg_cell_type'],</span> <span class="pre">by</span> <span class="pre">default</span></code></p></li>
<li><p>a UMAP layout for the query data in this reference embedding space - stored under <code class="docutils literal notranslate"><span class="pre">adata.obsm['cxg_cell_type_umap']</span></code> by default</p></li>
<li><p>metadata about the model used - stored under <code class="docutils literal notranslate"><span class="pre">adata.uns['cxg_cell_type_predictions']['model_url']</span></code> by default</p></li>
</ul>
<p>The <code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span></code> subcommand offers a number of options to customize how the workflow is run and to adapt to different scenarios. You can display the full list of available options like so:</p>
<p><code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">annotate</span> <span class="pre">-h</span></code></p>
</section>
<section id="step-2-explore-and-refine">
<h3>Step 2: Explore and Refine<a class="headerlink" href="#step-2-explore-and-refine" title="Permalink to this headline">#</a></h3>
<p>After you have successfully completed the annotation workflow, you are now ready to explore your results in CELLxGENE Annotate!</p>
<p>You can launch your annotated <code class="docutils literal notranslate"><span class="pre">anndata</span></code> object in the CELLxGENE Annotate like so:</p>
<p><code class="docutils literal notranslate"><span class="pre">cellxgene</span> <span class="pre">launch</span> <span class="pre">./pathTo/annotatedData.h5ad</span></code></p>
<p>You will now have access to the standard set of CELLxGENE Annotate features to explore your single cell dataset. Here are some initial steps that you might take to start exploring your annotated dataset:</p>
<ul class="simple">
<li><p>color by predicted cell type</p></li>
<li><p>color by prediction confidence score</p></li>
<li><p>calculate differential expression between two annotated populations to see which genes distinguish them from each other</p></li>
<li><p>create custom annotations to further refine or correct the predicted annotations</p></li>
</ul>
<p>For a reference on how one might refine predicted annotations, you can refer to this <a class="reference external" href="https://www.nature.com/articles/s41596-021-00534-0">Nature Tutorial</a>.</p>
</section>
<section id="useful-links">
<h3>Useful Links<a class="headerlink" href="#useful-links" title="Permalink to this headline">#</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/tutorial_ov">10X cellranger</a></p></li>
<li><p><a class="reference external" href="https://satijalab.org/seurat/">Seurat</a></p></li>
<li><p><a class="reference external" href="https://scanpy.readthedocs.io/en/stable/">ScanPy</a></p></li>
<li><p><a class="reference external" href="https://cellxgene.cziscience.com/docs/05__Annotate%20and%20Analyze%20Your%20Data/5_4__Annotating%20Data">annotation</a></p></li>
<li><p><a class="reference external" href="https://www.nature.com/articles/s41596-021-00534-0">Tutorial: guidelines for annotating single-cell transcriptomic maps using automated and manual methods</a></p></li>
</ul>
</section>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./annotate"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

            </article>
            

            
            
            <footer class="bd-footer-article">
                <!-- Previous / next buttons -->
<div class='prev-next-area'>
  <a class='left-prev' id="prev-link" href="annotate.html" title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
          <p class="prev-next-subtitle">previous</p>
          <p class="prev-next-title">Annotate Data</p>
      </div>
  </a>
  <a class='right-next' id="next-link" href="../census/census.html" title="next page">
  <div class="prev-next-info">
      <p class="prev-next-subtitle">next</p>
      <p class="prev-next-title">CZ CELLxGENE Discover Census</p>
  </div>
  <i class="fa-solid fa-angle-right"></i>
  </a>
</div>
            </footer>
            
          </div>
          
          
          
            <div class="bd-sidebar-secondary bd-toc">
              
<div class="toc-item">
  
<div class="tocsection onthispage">
    <i class="fa-solid fa-list"></i> On this page
</div>
<nav id="bd-toc-nav" class="page-toc">
    <ul class="visible nav section-nav flex-column">
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#">
   Predict
  </a>
 </li>
 <li class="toc-h1 nav-item toc-entry">
  <a class="reference internal nav-link" href="#automated-annotation-of-single-cell-data-experimental">
   Automated Annotation of Single Cell Data (EXPERIMENTAL)
  </a>
  <ul class="visible nav section-nav flex-column">
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#installation-of-cellxgene-annotate">
     Installation of CELLxGENE[annotate]
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#option-1-conda-setup">
       Option 1: conda setup
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#option-2-venv-setup">
       Option 2: venv setup
      </a>
     </li>
    </ul>
   </li>
   <li class="toc-h2 nav-item toc-entry">
    <a class="reference internal nav-link" href="#cellxgene-annotation-workflow">
     CELLxGENE Annotation Workflow
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#quickstart">
       Quickstart
      </a>
      <ul class="nav section-nav flex-column">
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#conda">
         conda
        </a>
       </li>
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#venv">
         venv
        </a>
       </li>
      </ul>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-0-annotation-workflow-inputs">
       Step 0: Annotation workflow inputs
      </a>
      <ul class="nav section-nav flex-column">
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#query-data">
         Query data
        </a>
       </li>
       <li class="toc-h4 nav-item toc-entry">
        <a class="reference internal nav-link" href="#prediction-model">
         Prediction model
        </a>
       </li>
      </ul>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-1-running-cellxgene-annotate">
       Step 1: Running
       <code class="docutils literal notranslate">
        <span class="pre">
         cellxgene
        </span>
        <span class="pre">
         annotate
        </span>
       </code>
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#step-2-explore-and-refine">
       Step 2: Explore and Refine
      </a>
     </li>
     <li class="toc-h3 nav-item toc-entry">
      <a class="reference internal nav-link" href="#useful-links">
       Useful Links
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
</ul>

</nav>
</div>

            </div>
          
          
        </div>
        <footer class="bd-footer-content">
          <div class="bd-footer-content__inner">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By CZI CELLxGENE team
</p>

  </div>
  
  <div class="footer-item">
    
<p class="copyright">

    &copy; Copyright 2023.<br>

</p>

  </div>
  
  <div class="footer-item">
    <p class="last-updated">
Last updated on None.<br>
</p>
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </div>
        </footer>
        

      </main>
    </div>
  </div>

  
    
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/bootstrap.js?digest=796348d33e8b1d947c94"></script>
<script src="../_static/scripts/pydata-sphinx-theme.js?digest=796348d33e8b1d947c94"></script>

  </body>
</html>